halo2_proofs/src/dev.rs

```diff
@@ -936,7 +959,7 @@ impl<F: FieldExt> MockProver<F> {
                                 .collect::<Vec<_>>()
                         })
                         .collect();
-                    self.usable_rows.clone().filter_map(move |input_row| {
+                    row_ids.into_iter().filter_map(move |input_row| {
```

Comment:
It looks like this also disables the verification of lookups to just the passed in `row_ids`, but the EVM circuit does do lookups on rows that don't have `q_step` enabled (currently just byte lookups, but in the future all kinds of lookups). If that's the case (and I could certainly be wrong) then I don't think we can limit the rows like this (at least not with the same `row_ids` as for the gates).

Reply:
ok i understand, seems it is better change method signature to `pub fn verify_at_rows<I: Iterator<Item = usize>>(
        &self,
        gate_row_ids: I,
        lookup_input_row_ids: I,
    )`